aerogear/mobile-security-service-operator
# feat(Use Route) : Replace ingress for route and change the related logic impl

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9106

## What
* Update de operator-sdk version
* Add route scheme
* Remove clusterHost impl in the specs
* Impl in the app Controller to get the route and check its host
* Impl changes in the impact code
* update the readme
* allow use specify router name by CR
* allow user specify platform/cluster used in order to create or not the route
* update the readme
* improvements and refactories 

## Verification Steps
 
1. Clone this PR and use the image `quay.io/aerogear/mobile-security-service-operator:0.1.0-dev`
2. Run `make run-local` to start the operator from the local source code
3. Check that the operator was installed with success 
4. Check that the router was created with success
5. Crate a new namespace ( oc new-project )
6. Create the app (make create-app)
7. Check that the app was created in the REST Service
8. Check that the configMap was created with the serviceURL
9. Run now `make unbind`
10. Check that thee configMap was removed and the app was soft deleted in the REST Service  

## Checklist:


## Progress


## Additional Notes

<img width="802" alt="Screenshot 2019-05-01 at 23 39 51" src="https://user-images.githubusercontent.com/7708031/57048184-e244be80-6c6a-11e9-8c2b-5774630939b6.png">
<img width="1598" alt="Screenshot 2019-05-01 at 23 39 44" src="https://user-images.githubusercontent.com/7708031/57048185-e244be80-6c6a-11e9-9716-608153878308.png">
<img width="1570" alt="Screenshot 2019-05-01 at 23 39 35" src="https://user-images.githubusercontent.com/7708031/57048186-e2dd5500-6c6a-11e9-9c65-ba92eb5947b1.png">
<img width="1631" alt="Screenshot 2019-05-01 at 23 39 29" src="https://user-images.githubusercontent.com/7708031/57048187-e2dd5500-6c6a-11e9-98c3-e51f046a3a8d.png">
<img width="1543" alt="Screenshot 2019-05-01 at 23 39 17" src="https://user-images.githubusercontent.com/7708031/57048188-e2dd5500-6c6a-11e9-98c8-075412882b16.png">
<img width="1609" alt="Screenshot 2019-05-01 at 23 39 09" src="https://user-images.githubusercontent.com/7708031/57048189-e2dd5500-6c6a-11e9-9809-be3530663321.png">
